search-service/src/main/java/de/cxp/ocs/elasticsearch/query/filter/TermResultFilter.java

```diff
@@ -14,29 +20,35 @@
 
 	private final Field field;
 
-	private final String[] values;
+	private final List<String> valuesAsList = new ArrayList<>();
 
 	private boolean filterOnId = false;
 
 	private String fieldPrefix = FieldConstants.TERM_FACET_DATA;
 
 	public TermResultFilter(Field field, String... inputValues) {
 		this.field = field;
-		values = inputValues;
+
+		for (String singleValue : inputValues) valuesAsList.add(singleValue);
 	}
 
 	public String getSingleValue() {
-		if (values.length > 0) return values[0];
+		if (valuesAsList.size() > 0) return valuesAsList.get(0);
 		return null;
 	}
 
 	public String getValue(int index) {
-		if (values.length > index) return values[index];
+		if (valuesAsList.size() > index) return valuesAsList.get(index);
 		return null;
 	}
 
 	@Override
 	public boolean isNestedFilter() {
 		return true;
 	}
+
+	@Override
+	public String[] getValues(){
+		return valuesAsList.stream().toArray(String[]::new);
```

Comment:
You can use `valueAsList.toArray(new String[0])`, which avoids interation and copies the internal array directly.

Reply:
ok